### ISSUE TITLE
make relocate! non-allocating

### ISSUE THREAD

**Issue by Datseris** [JuliaDynamics/DynamicalBilliards.jl#78]
```julia
tmin, i = next_collision(p, bt)

(0.388…, 2)
@btime relocate!($p, $bt[i], $tmin)

  59.723 ns (2 allocations: 32 bytes)
```
I think there is no reason for `relocate!` to allocate, besides bad programming.

**Comment by Datseris** [comment]
Hm I don't know where the allocation happens from. It is not from relocate?... I assume it happens when you do `bt[i]`.

```
@btime bounce!($p, $bt);

  119.237 ns (2 allocations: 32 bytes)

o = bt[3]
@btime resolvecollision!($p, $o)

  8.210 ns (0 allocations: 0 bytes)
```

**Comment by Datseris** [comment]
@lhupe can you write here your observations on why this is allocating?

**Comment by lhupe** [comment]
I played around with this behaviour earlier while working on #73, where I observed that accessing an obstacle in the billiard (eg. `bt[i]`) causes two allocations. As this happens exactly once in `bounce!`, it would explain the origin of the two allocations.

**Comment by lhupe** [comment]
`relocate!` by itself isn't allocating anything, if you move the obstacle access outside the benchmark, no allocations are reported.
``` julia
> tmin, i = next_collision(p, bt)
> o = bt[i]
> @btime relocate!($p, $o, $tmin)
 
32.623 ns (0 allocations: 0 bytes)
```

**Comment by Datseris** [comment]
I am closing this then for now. This issue is solved at Julia 0.7. When we update if we see a problem we can re-open.
